sycl/include/sycl/ext/oneapi/bindless_images.hpp

```diff
@@ -910,6 +910,22 @@ DataT fetch_image(const sampled_image_handle &imageHandle [[maybe_unused]],
 #endif
 }
 
+template <typename DataT>
+std::enable_if_t<std::is_same_v<DataT, float4> || std::is_same_v<DataT, int4> ||
+                     std::is_same_v<DataT, uint4>,
+                 DataT>
+gather_image(const sampled_image_handle &imageHandle [[maybe_unused]],
```

Comment:
Could we use some `[[__sycl_detail__::__uses_aspects__(aspect::something_here)]]` to ensure that it only ends up running on devices that support it? That should make the runtime throw an exception if the user tries to run it on anything else.

Reply:
Yeah, although I haven't actually added an aspect for this feature in this PR, because actually the team decided that we have too many aspects already, so what is being proposed is a more universal query interface that supports various different image options. I am not sure how your proposal/point would affect this. @ProGTX what do you think?

Reply:
If we otherwise prevent the images from being constructed for devices that don't support them, that should be fine too, as it prevents this function from being called on a valid object. Where the aspect-check becomes useful is in cases where the use of these functions may be in code that we can't prevent the user from using ahead of trying to launch the kernel.

Reply:
I will update the extension documentation to mention the new aspect once approvals are given.

Reply:
> I will update the extension documentation to mention the new aspect once tests pass and approvals are given.

Done